### DISCUSSION TITLE
OOMKilled containers and memory_limiter

### DISCUSSION THREAD

**Discussion by vinzent** [signalfx/splunk-otel-collector-chart#903] [General]
We see regularly some pods being killed by the OOMKiller. I see that the helm-chart configures memory_limiter limit_mib based on the memory limit of the container.

I was digging a little bit into and found this in the memory_limiter docs:

> limit_mib: Maximum amount of memory, in MiB, targeted to be allocated by the process heap. Note that typically the total memory usage of process will be about 50MiB higher than this value. This defines the hard limit.

(Ref: https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiterprocessor/README.md)

So this means, that it is expected that the process can use more than the limit_mib and thus more than the container limit. Which in turn leads to the container being OOM killed.

**Comment by atoulme** [comment]
Please get in touch with a support case so we may discuss the specifics of your situation. Please note the way memory is handled by the collector is under discussion with https://github.com/open-telemetry/opentelemetry-collector/issues/8343.
